setup.cfg

```diff
@@ -58,19 +61,18 @@ exclude =
 numba_extensions =
     init = awkward._connect._numba:register
 
-
 [tool:pytest]
 norecursedirs = src pybind11 rapidjson dependent-project studies
 
 [flake8]
-select = C,E,F,W,B,B9,T
 # F401 can be fixed by adding __all__'s
 # E712 should be dropped, False/True should compared by "is" or checked as booleans, but not "=="
 # E711 should be dropped, always compare None with is, not ==.
 # F841 should be removed, use nothing or _ or _<name> for an unused variable
 # E402 can be improved in the future with isort
 # F403 should be removed, import * is evil
 # B007 should be removed, variables that are not used should start with _
```

Comment:
I'll investigate the reasons for these and perhaps lift some of them.

Reply:
Oh wait—I have to fix these before work on setup.cfg can be completed? In that case, I'll do it now.

Reply:
No, you don't have to, just once they are cleared, we can enable automatic formatting here and forget about it. That also includes some nice consistency checks.

When you add `__all__`, I would also add `def __dir__(): return __all__`, as that really nicely cleans up tab completion in REPLs like IPython.

Reply:
Well, I already started and code-cleanup is a good thing, anyway.

I'll add a `__dir__` for each `__all__`.